Classes/GarbageCollector.php

```diff
@@ -215,6 +219,33 @@ protected function deleteIndexDocuments($table, $uid)
     }
 
     /**
+     * Delete all index documents that are related to a TYPO3 "page"
+     *
+     * @param $uid
+     */
+    protected function deleteRecordsOfPage($uid)
+    {
+        /** @var \ApacheSolrForTypo3\Solr\ConnectionManager $connectionManager */
+        $connectionManager = GeneralUtility::makeInstance('ApacheSolrForTypo3\\Solr\\ConnectionManager');
+        $indexQueue = GeneralUtility::makeInstance('ApacheSolrForTypo3\\Solr\\IndexQueue\\Queue');
+        $site = Site::getSiteByPageId($uid);
+
+        // a site can have multiple connections (cores / languages)
+        $solrConnections = $connectionManager->getConnectionsBySite($site);
+
+        /** @var  $solr */
+        foreach ($solrConnections as $solr) {
+            $response = $solr->search('pid:' . intval($uid), 0, 10000);
```

Comment:
You must/should not use Solr to find the documents that need to be deleted. The source of truth is the Index Queue and/or the TYPO3 DB in general. So you'll have to find the items there and then delete from Solr using that information.

You then just call `deleteIndexDocuments()` and don't iterate over the Solr connections here.